File: ZemljaSlova.Model/Reports/BooksSoldReport.cs
using System;
using System.Collections.Generic;

namespace ZemljaSlova.Model.Reports
{
    public class BooksSoldReport
    {
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public int TotalBooksSold { get; set; }
        public decimal TotalRevenue { get; set; }
        public int TotalTransactions { get; set; }
        public List<BookSoldTransaction> Transactions { get; set; } = new List<BookSoldTransaction>();
        public List<BookSoldSummary> BookSummaries { get; set; } = new List<BookSoldSummary>();
        public string ReportPeriod { get; set; } = string.Empty;
    }

    public class BookSoldTransaction
    {
        public int Id { get; set; }
        public string BookTitle { get; set; } = string.Empty;
        public string AuthorNames { get; set; } = string.Empty;
        public int Quantity { get; set; }
        public decimal UnitPrice { get; set; }
        public decimal TotalPrice { get; set; }
        public DateTime SoldDate { get; set; }
        public string EmployeeName { get; set; } = string.Empty;
        public string CustomerName { get; set; } = string.Empty;
    }

    public class BookSoldSummary
    {
        public int BookId { get; set; }
        public string BookTitle { get; set; } = string.Empty;
        public string AuthorNames { get; set; } = string.Empty;
        public int TotalQuantitySold { get; set; }
        public decimal TotalRevenue { get; set; }
        public decimal AveragePrice { get; set; }
    }
}

File: UI/zemljaslova_desktop/lib/models/reports/books_sold_report.dart
import 'package:json_annotation/json_annotation.dart';

part 'books_sold_report.g.dart';

@JsonSerializable()
class BooksSoldReport {
  @JsonKey(name: 'startDate')
  final DateTime startDate;
  
  @JsonKey(name: 'endDate')
  final DateTime endDate;
  
  @JsonKey(name: 'totalBooksSold')
  final int totalBooksSold;
  
  @JsonKey(name: 'totalRevenue')
  final double totalRevenue;
  
  @JsonKey(name: 'totalTransactions')
  final int totalTransactions;
  
  @JsonKey(name: 'transactions')
  final List<BookSoldTransaction> transactions;
  
  @JsonKey(name: 'bookSummaries')
  final List<BookSoldSummary> bookSummaries;
  
  @JsonKey(name: 'reportPeriod')
  final String reportPeriod;

  BooksSoldReport({
    required this.startDate,
    required this.endDate,
    required this.totalBooksSold,
    required this.totalRevenue,
    required this.totalTransactions,
    required this.transactions,
    required this.bookSummaries,
    required this.reportPeriod,
  });

  factory BooksSoldReport.fromJson(Map<String, dynamic> json) => _$BooksSoldReportFromJson(json);
  Map<String, dynamic> toJson() => _$BooksSoldReportToJson(this);
}

@JsonSerializable()
class BookSoldTransaction {
  final int id;
  
  @JsonKey(name: 'bookTitle')
  final String bookTitle;
  
  @JsonKey(name: 'authorNames')
  final String authorNames;
  
  final int quantity;
  
  @JsonKey(name: 'unitPrice')
  final double unitPrice;
  
  @JsonKey(name: 'totalPrice')
  final double totalPrice;
  
  @JsonKey(name: 'soldDate')
  final DateTime soldDate;
  
  @JsonKey(name: 'employeeName')
  final String employeeName;
  
  @JsonKey(name: 'customerName')
  final String customerName;

  BookSoldTransaction({
    required this.id,
    required this.bookTitle,
    required this.authorNames,
    required this.quantity,
    required this.unitPrice,
    required this.totalPrice,
    required this.soldDate,
    required this.employeeName,
    required this.customerName,
  });

  factory BookSoldTransaction.fromJson(Map<String, dynamic> json) => _$BookSoldTransactionFromJson(json);
  Map<String, dynamic> toJson() => _$BookSoldTransactionToJson(this);
}

@JsonSerializable()
class BookSoldSummary {
  @JsonKey(name: 'bookId')
  final int bookId;
  
  @JsonKey(name: 'bookTitle')
  final String bookTitle;
  
  @JsonKey(name: 'authorNames')
  final String authorNames;
  
  @JsonKey(name: 'totalQuantitySold')
  final int totalQuantitySold;
  
  @JsonKey(name: 'totalRevenue')
  final double totalRevenue;
  
  @JsonKey(name: 'averagePrice')
  final double averagePrice;

  BookSoldSummary({
    required this.bookId,
    required this.bookTitle,
    required this.authorNames,
    required this.totalQuantitySold,
    required this.totalRevenue,
    required this.averagePrice,
  });

  factory BookSoldSummary.fromJson(Map<String, dynamic> json) => _$BookSoldSummaryFromJson(json);
  Map<String, dynamic> toJson() => _$BookSoldSummaryToJson(this);
}
